src/main/kotlin/app/cash/licensee/plugin.kt

```diff
@@ -162,6 +166,26 @@ private fun configureAndroidVariants(
     rootTask.configure {
       it.dependsOn(task)
     }
+
+    if (extension.enableAndroidAssetGeneration.get()) {
+      val capitalizedVariantName = variant.name.replaceFirstChar { it.titlecase(ROOT) }
+      val copyArtifactsTask =
+        project.tasks.register(
+          "copy${capitalizedVariantName}LicenseeReportToAssets",
+          AssetCopyTask::class.java,
+        ) { asset ->
+          asset.targetFileName.set("artifacts.json")
+          project.tasks.withType(LicenseeTask::class.java)
+            .findByName("licenseeAndroid$capitalizedVariantName")?.let {
+              asset.inputFile.set(it.jsonOutput)
+            }
```

Comment:
Calls to `findByName` will realize all tasks in the collection. I think this can just be
```suggestion
          asset.inputFile.set(task.flatMap { it.jsonOutput })
```

Reply:
yep, just that worked.